lib/order.rb

```diff
@@ -0,0 +1,99 @@
+require 'csv'
+require_relative 'customer'
+
+class Order
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+
+  @@all_orders = []
+
+  # Constructor method to initialize all Order.new instances
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id.to_i
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+
+    # Raise ArgumentError if fulfillment_status is not one of five options
+    @fulfillment_options = [:pending, :paid, :processing, :shipped, :complete]
+    unless @fulfillment_options.include? @fulfillment_status
+      raise ArgumentError, 'Fulfillment status must be :pending, :paid, :processing, :shipped, or :complete. '
+    end
+  end
+
+  # Calculate total of Order instance products
+  def total
+    # if @products.empty, product sum is 0
+    if @products == {} || @products == 0
+      @product_sum = 0
+    else
+      # @product_sum = @products.values.sum
+      total = 0
+      @products.each_value do |value|
+        total += value
+        @product_sum = (total * 1.075).round(2)
+      end
+    end
```

Comment:
You shouldn't need to store the sum in an instance variable here, since you're not using it elsewhere in this class. Instead you should use a local variable.